plugin/kubernetes/kubernetes.go

```diff
@@ -455,6 +469,11 @@ func (k *Kubernetes) findServices(r recordRequest, zone string) (services []msg.
 							s := msg.Service{Host: addr.IP, Port: int(p.Port), TTL: k.ttl}
 							s.Key = strings.Join([]string{zonePath, Svc, svc.Namespace, svc.Name, endpointHostname(addr, k.endpointNameMode)}, "/")
 
+							if _, dup := uniq[s]; dup {
```

Comment:
unclear why this is needed and I think we _alread_ remove duplicates from the answer

Reply:
IIRC, the backend does duplicate removal for some query types, but not all (e.g. not for A/AAAA).

I added de-dup (where not already present) and tests for EndpointSlice skew address duplication. With EndpointSlices, duplication of an Endpoint's Address is possible since we don't receive all EndpointSlices for any given Service at one time. https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#duplicate-endpoints

Reply:
the dedup exists in backend.go (other discussion if that is still useful), but we should add it there, not here.
Thanks the ref to those docs

Reply:
Yes, the backend already does de-dup for all types that I can see. I was staring right at it and didn't see it.  I've removed all de-dup from this PR.